operations/helm/charts/alloy/templates/extra-manifests.yaml

```diff
@@ -1,4 +1,15 @@
-{{ range .Values.extraObjects }}
+{{- /* Normalize extraObjects to a list, easier to loop over */ -}}
+{{- $extraObjects := .Values.extraObjects | default (list) -}}
+
+{{- if kindIs "map" $extraObjects -}}
+  {{- $extraObjects = values $extraObjects -}}
+{{- end -}}
+
+{{- range $extraObjects }}
 ---
-{{ tpl (toYaml .) $ }}
-{{ end }}
+  {{- if kindIs "map" . }}
+    {{- tpl (toYaml .) $ | nindent 0 }}
+  {{- else if kindIs "string" . }}
```

Comment:
Is it possible that it's neither a map nor a string? E.g. could it contain numbers?

Reply:
It could, and we could template that, but it would yield invalid kube yaml so ultimately would be rejected.  The user does need to define multiple fields, at least a minimum to render valid yaml.  As far as I can think, these are the only two object types that would allow defining a complete object.